### PR TITLE
Feat/add file uploader

### PR DESCRIPTION
**Pull request by CorneeSean**
This PR adds a UI component for FileUploading.
It does not react on form submit event and does not send files to the backend - this will be added in one of next PR.

@chodorowicz That's a long one, take your time ;)

### CODE REVIEW

**Comment by CorneeSean** [comment]
The tests are failing, I will fix it :)

File: frontend/src/app/modules/intervention-form/file-uploader/file-uploader.component.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        ),
+    );
+  }
+
+  onFileRemove(removedFile: File, input: HTMLInputElement) {
+    this.files = this.files.filter(file => file !== removedFile);
+    input.value = '';
+  }
+
+  getAcceptedFileTypes(): string {
```
Comment (CorneeSean) [inline frontend/src/app/modules/intervention-form/file-uploader/file-uploader.component.ts:55]: To slightly improve performance, this needs to be changed to an instance field and initialized during ngOnInit

File: frontend/src/app/modules/intervention-form/file-uploader/file-uploader.component.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  openFileBrowser(input: HTMLInputElement) {
+    input.click();
+  }
+
+  onFilesSelected(input: HTMLInputElement) {
+    const newFiles = this.filterExistingFiles(input.files, this.files);
+    this.files = this.files.concat(newFiles);
+    input.value = '';
```
Comment (chodorowicz) [inline frontend/src/app/modules/intervention-form/file-uploader/file-uploader.component.ts:37]: Do we need to clear it? In general manipulating input.value of file input doesn't have any effect due to security restrictions.

Comment (CorneeSean) [reply frontend/src/app/modules/intervention-form/file-uploader/file-uploader.component.ts:37]: I believe you are partially right - empty string is the only allowed value for the input.value manipulated from javascript, as far as I remember.
However, with current implementation of the component it does not have any effect on its workings.
I am pretty sure there was a time when I needed it during prototyping phase, but sincerely I do not remember why, as this was more than a month ago :)

I will remove it, good catcht!

**Pull request merged**